Presentation: Improve presentation set up learning page (#2156)
imodeljs/imodeljs: docs/presentation/Setup/index.md

# Setting Up iTwin.js Presentation Library

> Before the Presentation library can be used, it needs to be properly initialized. Because of the nature of [iTwin.js architecture](../../SoftwareArchitecture.md), an iTwin.js application always has a backend and may or may not have a frontend, which means the library needs to be initialized on the backend and on the frontend if it is used - just like the iTwin.js framework itself.

Reviewer: ```suggestion
Before the Presentation library can be used, it needs to be properly initialized. Because of the nature of [iTwin.js architecture](../../SoftwareArchitecture.md), the library needs to be initialized on the backend and each frontend that is in use — just like the iTwin.js framework itself.
```

   ``` ts
   [[include:Presentation.Common.RpcInterface.Imports]]
   [[include:Presentation.Common.RpcInterface]]
   [[include:Presentation.Frontend.RpcInterface.Options]]
   [[include:Presentation.Frontend.IModelAppStartup]]
   ```

   **Note:** The above example uses `ElectronApp`, but it's similar with other specializations like `WebViewerApp`.

2. Initialize Presentation frontend:

   ``` ts
   [[include:Presentation.Frontend.Imports]]
   [[include:Presentation.Frontend.Initialization]]
   ```

## Next Steps

After initializing the Presentation library on both backend and frontend, the library is ready to use. Check out the following for examples:

- Blogs
  - [Demystifying Unified Selection and Selection Scopes](https://medium.com/itwinjs/demystifying-unified-selection-and-selection-scopes-88b9418c57f3)
  - [Hooking 3rd Party Component into Unified Selection](https://medium.com/itwinjs/hooking-3rd-party-component-into-unified-selection-c4daec69789d)
- Samples
  - [A simple Presentation-driven hierarchy](https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=presetation-tree-sample)

>   - [Retrieving properties](https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=property-formatting-sample)

Author: Agree that it's not the right place.

My suggestion would be to create a separate "External Resources" or "Examples" section in Content, Hierarchies, Customization and Unified Selection pages and add appropriate links there. A link to examples could be added to each category in the sidebar as well. How does that sound?

>   - [Retrieving properties](https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=property-formatting-sample)

Reviewer: `External Resources` sections would be sufficient.

>   - [Retrieving properties](https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=property-formatting-sample)

Author: done
